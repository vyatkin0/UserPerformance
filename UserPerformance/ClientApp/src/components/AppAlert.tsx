/**
 * Content and markup of small Alert window with information about results of latest actions
 */

import Alert, {Color as AlertColor} from '@material-ui/lab/Alert';
import React from 'react';
import ReactDOM from 'react-dom'

import { WithStyles, withStyles } from '@material-ui/core/styles';

import styles from '../styles';

export let currentMessageId: number = 0;

export interface InfoMessage {
   id: number;
   severity: AlertColor;
   text: string;
   timeoutID?: any;
   onClick?: Function;
}

interface AppAlertProps extends WithStyles<typeof styles> {
    messages: InfoMessage[];
}

class AppAlert extends React.PureComponent<AppAlertProps> {
    private handleClickAlert = (id: number) => {
        if(this.props.messages.length < 1) {
            return null;
        }

        const index = this.props.messages.findIndex(m => m.id===id);

        if (index > -1) {

            if(this.props.messages[index].timeoutID){
                clearTimeout(this.props.messages[index].timeoutID);
            }

            if(this.props.messages[index].onClick){
                this.props.messages[index].onClick();
            }

            this.props.messages.splice(index, 1);
            this.forceUpdate();
        }
    }

    public componentWillUnmount() {   
        this.props.messages.forEach( m =>{
            if(m.timeoutID) {
                clearTimeout(m.timeoutID);
                m.timeoutID = null;
            }
        });
    }

    private onClickAlert = () => this.handleClickAlert(this.props.messages[0].id);

    public render() {
        if(this.props.messages.length < 1) {
            return null;
        }

        const { classes } = this.props;
        
        const message = this.props.messages[0];

        if(message.severity === 'success'){
            message.timeoutID = setTimeout(()=>this.handleClickAlert(message.id), 3000);
        }
        
        return ReactDOM.createPortal(<Alert className={classes.alert} variant='filled' severity={message.severity} onClick={this.onClickAlert}>
                {message.text}
            </Alert>,
             document.getElementById("bottomAlert")) 
    }
}

export default withStyles(styles)(AppAlert);