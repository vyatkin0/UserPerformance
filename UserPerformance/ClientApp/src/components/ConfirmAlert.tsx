/**
 * Confirmation alert for delete activity action
 */

import * as React from "react";

import {
    Button,
    Dialog,
    DialogActions,
    DialogContent,
    DialogContentText,
    DialogTitle,
} from "@material-ui/core";

interface ConfirmAlertProps {
    open: boolean;
    title: string;
    message: string | JSX.Element;
    handleYes: () => void;
    handleNo: () => void;
}

export default class ConfirmAlert extends React.PureComponent<ConfirmAlertProps> {
    private handleClose = (yes: boolean) => {
        if (yes) {
            return this.props.handleYes();
        }

        this.props.handleNo();
    };

    public render() {
        if (!this.props.open) {
            return null;
        }

        return <Dialog
                open={this.props.open}
                onClose={this.handleClose}
                aria-labelledby="alert-dialog-title"
                aria-describedby="alert-dialog-description"
            >
                <DialogTitle id="alert-dialog-title">
                    {this.props.title}
                </DialogTitle>
                <DialogContent>
                    <DialogContentText id="alert-dialog-description">
                        {this.props.message}
                    </DialogContentText>
                </DialogContent>
                <DialogActions>
                    <Button
                        onClick={() => this.handleClose(false)}
                        color="primary"
                        autoFocus
                    >
                        No
                    </Button>
                    <Button
                        onClick={() => this.handleClose(true)}
                        color="primary"
                    >
                        Yes
                    </Button>
                </DialogActions>
            </Dialog>
    }
}
