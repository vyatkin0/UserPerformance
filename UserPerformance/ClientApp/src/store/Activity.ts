/**
 * Activity object interface
 */
export default interface Activity {
    id?: number; // Identifier in Database
    parentId?: number; // Parent identifier
    name: string; // Name of activity
    description: string; // Description of activity
    workCost?: number; // Spent time for activity
}