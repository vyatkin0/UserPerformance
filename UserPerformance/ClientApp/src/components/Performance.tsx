/**
 * Content and markup of Performance page
 */

import * as PerformanceStore from '../store/Performance';
import * as React from 'react';
import * as ReactDOM from 'react-dom'

import {
    Button,
    Popover,
    Table,
    TableBody,
    TableCell,
    TableContainer,
    TableHead,
    TableRow,
    TextField
} from '@material-ui/core';
import { WithStyles, withStyles } from '@material-ui/core/styles';

import AppAlert from './AppAlert'
import { ApplicationState } from '../store';
import Paper from '@material-ui/core/Paper';
import PeriodSelect from './PeriodSelect';
import { connect } from 'react-redux';
import styles from '../styles';

type PerformanceProps =  PerformanceStore.PerformanceState
 & WithStyles<typeof styles>
 & typeof PerformanceStore.actionCreators; // ... plus action creators we've requested

class Performance extends React.PureComponent<PerformanceProps> {

    constructor(props) {
        super(props);
        this.countsInputRef = React.createRef(); // Input in popover window 
    }

    public componentDidMount() {
        
        // Support for electron environment
        const w = window as any;
        if (typeof w.ipcRenderer === 'undefined') {
            window.addEventListener('beforeunload', this.confirmExit.bind(this));
        } else {
            w.ipcRenderer.on('app-close', (event, arg) => {
                const fakeEvent = {returnValue: true, preventDefault:()=>{}};
                this.confirmExit(fakeEvent);

                event.returnValue = typeof fakeEvent.returnValue !== 'undefined';
                w.ipcRenderer.send('closed', event.returnValue);
            });
        }
 
        const now = new Date();
        const from = new Date(now.getFullYear(), now.getMonth(), now.getDate());
        const to = new Date(now.getFullYear(), now.getMonth(), now.getDate() + 1);
        this.props.requestPerformance(from, to, true);
        //this.ensureDataFetched();
    }
    
    public componentDidUpdate() {
        this.ensureDataFetched();
    }

    private ensureDataFetched() {
        this.props.requestPerformance(this.props.from, this.props.to, false);
    }

    private countsInputRef;

    private currentPopper = {
        open:false,
        anchorEl: null,
        activityId:undefined,
        dayIndex:undefined,
        countsHelper:'',
    };

    private confirmExit = (e) => {
        if(Object.keys(this.props.changes).length>0){
            // Cancel the event
            e.preventDefault(); // If you prevent default behavior in Mozilla Firefox prompt will always be shown

            // Chrome requires returnValue to be set
            e.returnValue = 'Changes are not saved!';

            return e.returnValue;
        }
        else {
            // the absence of a returnValue property on the event will guarantee the browser unload happens
            delete e['returnValue'];
        }
    }
    private handleClick = (event: React.MouseEvent<HTMLSpanElement>, activityId:number, dayIndex:number) => {
        if( this.props.performance.currentDay < 0) {
            return;
        }
        
        const offset = this.props.performance.currentDay-dayIndex;
        if( offset<0 || offset>=this.props.performance.editedDays ) {
            return;
        }

        this.currentPopper = {
            open: true,
            anchorEl: event.currentTarget,
            activityId,
            dayIndex,
            countsHelper:'',
        };

        this.forceUpdate();
    };

    private handleClose = (event, update: boolean) => {
        event.preventDefault();

        const {activityId, dayIndex} = this.currentPopper;
        
        if ( typeof activityId === 'undefined'
            || activityId === null) return;

        if(update) {
            if(0 === activityId) {
                update = this.props.performance.days[dayIndex].hours != this.countsInputRef.current.value;
            } else {
                const userActivity = this.props.performance.userActivities.find(ea => ea.activity.id === activityId);

                update = userActivity && (userActivity.counts[dayIndex] != this.countsInputRef.current.value);
            }

            if(update && !this.validateFields(this.countsInputRef.current.value)) {
                this.forceUpdate();
                return;
            }

            this.props.updateActivityCount(activityId, dayIndex, this.countsInputRef.current.value);
        }

        this.currentPopper = {
            open: false,
            anchorEl: null,
            activityId:undefined,
            dayIndex:undefined,
            countsHelper:'',
        };

        this.forceUpdate();
    };
    
    private validateFields = (counts:string) => {

        const wrongValue = 'Wrong value';

        const n = Number(counts.replace(',','.'));
        let result = isFinite(n);

        if(result) {
            if(this.currentPopper.activityId===0) {
                if(n>24) {
                    result = false;
                    this.currentPopper.countsHelper = 'More than 24 hours per day';
                }
            }

            if (n<0) {
                result = false;
                this.currentPopper.countsHelper = wrongValue;
            }
        }
        else {
            // Для отработанных часов в неделю canExpand===false
            if(this.currentPopper.activityId===0) {
                const allowedChars = ['V','H'];
                result = allowedChars.includes(counts.trim().toUpperCase());
            }

            if(!result) {
                this.currentPopper.countsHelper = wrongValue;
            }
        }

        return result;
    }

    private handlePopperKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
        switch(event.key)
        {
        case 'Enter':
            event.preventDefault();
            this.handleClose(event, true);
            break;
        }
    }

    private renderStub = () => 
        <div className='flex_container'>
            <div>
                <h1>Activities management</h1>
            </div>
        </div> 


    /**
     * Content and markup small windows for edit counts of task and day hours
     */
    private renderEditPopover = () => {

        const { classes } = this.props;
        const currentPopper = this.currentPopper;

        return <Popover
            id='simplePopper'
            open={currentPopper.open}
            anchorEl={currentPopper.anchorEl}
            onClose={(e, reason) => this.handleClose(e, reason==='backdropClick')}
            anchorOrigin={{
                vertical: 'top',
                horizontal: 'left',
            }}
            transformOrigin={{
                vertical: 'top',
                horizontal: 'left',
            }}>
            <div className={classes.popover}>
                {currentPopper.anchorEl && <TextField variant='outlined'
                    autoFocus
                    inputRef={this.countsInputRef}
                    label='Value'
                    defaultValue={currentPopper.anchorEl.innerText}
                    margin='dense'
                    InputProps={{
                        classes: {
                            input: classes.textField,
                        }}}
                    error={Boolean(currentPopper.countsHelper)}
                    helperText = {currentPopper.countsHelper}
                    onKeyPress =  {this.handlePopperKeyPress}
                />}
                {Boolean(currentPopper.countsHelper) && <div><Button autoFocus variant='outlined' size='small' onClick={(e) =>this.handleClose(e, false)}>
                    Close
                </Button></div>}
            </div>
            </Popover>
    }

    /**
     * Content and markup of summary for month performance
     */
    private renderMonthInfo = () => {
        const performance = this.props.performance;

        const now = new Date();
        now.setHours(0, 0, 0, 0);
        const nowNextDay = new Date(now);
        nowNextDay.setDate(nowNextDay.getDate() + 1);
        
        const lastDay = new Date(this.props.to);
        lastDay.setDate(lastDay.getDate() - 1);
        
        return <>
        <div className='flex_container'>
            <div>
                <h1>Activities management</h1>
                <h3>{performance.userName}</h3>
            </div>
            <PeriodSelect
            from = {this.props.from}
            to = {this.props.to}
            minYear = {this.props.performance.minYear}
            maxYear = {this.props.performance.maxYear}
            onChange = {(from: Date, to: Date)=> this.props.requestPerformance(from, to, false)}
            />
        </div>
        <div className='performance_container'>
            Month performance
            <span className={(performance.monthPerformance < 100 ? 'warning ' : '') + 'performance_data'} >{performance.monthPerformance+'%'}</span>
            Working days
            <span className='performance_data'>{performance.monthWorkDays}</span>
        </div>
        </>
    }

    /**
     * Content and markup of table with performance for each day of specified period
     */
    private renderDaysInfo = () => {

        const { classes } = this.props;

        const performance = this.props.performance;
        const handleClick = this.handleClick.bind(this);
        
        return <TableContainer component={Paper}>
            <Table className={classes.table} size='small' aria-label='simple table'>
                <TableHead>
                <TableRow>
                    <TableCell align='left'>Name</TableCell>
                    <TableCell className={classes.monthHeadCell} align='center'></TableCell>
                    {performance.days.map((day,index) =>
                        <TableCell key={'month:'+ index} className={classes.numCell} align='center'>{`${day.day.getDate()}.${day.day.getMonth()+1}`}</TableCell>)}
                </TableRow>
                </TableHead>
                <TableBody>
                <TableRow>
                    <TableCell component='th' scope='row'>Day performance</TableCell>
                    <TableCell align='center'></TableCell>
                    {performance.days.map((d, index)=>{

                        let monthPerformance = '';
                        const dayHours = Number(d.hours);

                        let className = null;

                        if(dayHours>0)
                        { 
                            const mp = performance.userActivities.reduce((result, a)=> {
                                if(a.counts[index]>0 && a.activity.workCost>0) {
                                    return result + a.counts[index]*a.activity.workCost/60/dayHours;
                                }
                                
                                return result;
                            } , 0);

                            monthPerformance = parseFloat((mp * 100).toFixed(2)) + '%'

                            if(mp<1) {
                                className = classes.warnCell;
                            }
                        }

                        return <TableCell key={'month:'+ index} align='center' className={className}>{monthPerformance}</TableCell>})}
                </TableRow>
                <TableRow>
                    <TableCell component='th' scope='row'>Spent hours</TableCell>
                    <TableCell align='center'></TableCell>
                    {performance.days.map((d, index)=> {
                        let className = index<=performance.currentDay && index+performance.editedDays>performance.currentDay ? classes.cellSpan + ' ' + classes.cellSpanCurrent : classes.cellSpan;

                        if(this.isValueChanged(d.day.valueOf(), 0)) {
                            className += ' ' + classes.changedValue;
                        }

                        return <TableCell key={'month:'+ index} align='center'>
                        {<span className={className} onClick={(e)=>handleClick(e, 0, index)}>{d.hours}</span>}
                    </TableCell>})}
                </TableRow>
                </TableBody>
            </Table>
        </TableContainer>
    }

    /**
     * Footer content and markup
     */
    private renderFooterPortal = () => {
        const { classes } = this.props;
        
        const saveBtnDisabled = !(Object.keys(this.props.changes).length>0);

        return ReactDOM.createPortal(<div className={classes.divButtons}>
            <Button variant='contained' color='secondary' disabled={saveBtnDisabled} onClick={this.props.sendPerformance}>
            Save
            </Button>
            <Button variant='contained' color='primary' href='#/activities'/*this.props.gotoActivities*/>
            Select activities
            </Button>
            </div>, document.getElementById('footerToolbar'))
    }

    /**
     * User Performance content and markup
     */
    private renderUserActivities = () => {

        const performance = this.props.performance;

        const { classes } = this.props;

        const groupedActivities = performance ? performance.groupedActivities : {};

        const handleClick = this.handleClick.bind(this);

        return Object.keys(groupedActivities).map(key => {
            if ('null' === key) {
                return null;
            }

            const parent = performance.userActivities.find(ua => ua.activity.id === Number(key)).activity;

            return <React.Fragment key={key}>
                <h2>{parent.name}</h2>
                <TableContainer component={Paper}>
                    <Table className={classes.table} size='small' aria-label='simple table'>
                        <TableHead>
                            <TableRow>
                                <TableCell align='left'>Activity</TableCell>
                                <TableCell className={classes.monthHeadCell} align='center'>Spent time</TableCell>
                                <TableCell className={classes.monthHeadCell} align='center'>Count per month</TableCell>
                                {performance.days.map((day, index) =>
                                    <TableCell key={key + ':' + index} className={classes.numCell} align='center'>{`${day.day.getDate()}.${day.day.getMonth() + 1}`}</TableCell>)}
                            </TableRow>
                        </TableHead>
                        <TableBody>
                            {groupedActivities[key].map(ua => <TableRow key={ua.activity.id}>
                                <TableCell component='th' scope='row'>{ua.activity.name}</TableCell>
                                <TableCell align='center'>{ua.activity.workCost}</TableCell>
                                <TableCell align='center'>{ua.countPerMonth}</TableCell>
                                {performance.days.map((d, index) => {
                                    let className = index <= performance.currentDay && index + performance.editedDays > performance.currentDay ? classes.cellSpan + ' ' + classes.cellSpanCurrent : classes.cellSpan;

                                    if (this.isValueChanged(d.day.valueOf(), ua.activity.id)) {
                                        className += ' ' + classes.changedValue;
                                    }

                                    return <TableCell key={ua.activity.id + ':' + index} align='center'>
                                        {<span className={className} onClick={(e) => handleClick(e, ua.activity.id, index)}>{ua.counts[index]}</span>}
                                    </TableCell>
                                })}
                            </TableRow>)}
                        </TableBody>
                    </Table>
                </TableContainer>
            </React.Fragment>
        })
    }

    isValueChanged(day, activityId) :boolean {
        if (Object.keys(this.props.changes).length<1) {
            return false;
        }

        return this.props.changes[day]
            && this.props.changes[day][activityId];
    }
    
    public render() {
        const performance = this.props.performance;
        
        if(!performance || !performance.days || performance.days.length<1) {
            return this.renderStub();
        }

        return (
            <>
                {this.renderMonthInfo()}
                {this.renderEditPopover()}
                {this.renderDaysInfo()}
                {this.renderUserActivities()}
 
                {this.props.messages.length>0 && <AppAlert messages={this.props.messages}/>}

                { this.renderFooterPortal() }
            </>
        );
    }
}

export default connect(
    (state: ApplicationState) => state.performance,
    PerformanceStore.actionCreators
  )(withStyles(styles)(Performance) as any);