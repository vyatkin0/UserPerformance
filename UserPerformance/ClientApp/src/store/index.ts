/**
 * Application store
 */

import * as Activities from './Activities';
import * as Performance from './Performance';

// The top-level state object
export interface ApplicationState {
    performance: Performance.PerformanceState | undefined;
    activities: Activities.ActivitiesState | undefined;
}

// Whenever an action is dispatched, Redux will update each top-level application state property using
// the reducer with the matching name. It's important that the names match exactly, and that the reducer
// acts on the corresponding ApplicationState property type.
export const reducers = {
    performance: Performance.reducer,
    activities: Activities.reducer
};

// This type can be used as a hint on action creators so that its 'dispatch' and 'getState' params are
// correctly typed to match your store.
export interface AppThunkAction<TAction> {
    (dispatch: (action: TAction) => void, getState: () => ApplicationState): void;
}
