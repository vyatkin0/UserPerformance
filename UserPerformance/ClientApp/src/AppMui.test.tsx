/**
 * Test for components with material-ui 
 */

/**
 * @jest-environment jsdom
 */

/**
  * https://gist.github.com/gaearon/e7d97cdf38a2907924ea12e4ebdf3c85
  * Or, to select the node environment globally, use this in your package.json:

  {
    "name": "my-project",
    "jest": {
      "testEnvironment": "node"
    }
  }
  The warning about useLayoutEffect should now disappear because window is no longer defined during test execution.
*/

import * as PerformanceStore from './store/Performance';
import * as React from "react";

import Adapter from "enzyme-adapter-react-16";
import AppAlert from "./components/AppAlert";
import ConfirmAlert from "./components/ConfirmAlert";
import { MemoryRouter } from 'react-router-dom';
import Performance from './components/Performance';
import { Provider } from 'react-redux';
import { configure } from "enzyme";
import { createMount } from "@material-ui/core/test-utils";

configure({ adapter: new Adapter() });
describe("Material UI tests", () => {
    const timeNow = new Date('2020-01-01');
    const realDate = Date;

   beforeAll(() => {
    (global as any).Date = class {
            public static now() {
                return timeNow.valueOf();
            }

            constructor(a,b,c) {
                if(a!==undefined && b!==undefined && c!==undefined) {
                    this._date = new realDate(a,b,c);
                } else if(a!==undefined) {
                    this._date = new realDate(a);
                }

                if(!this._date) {
                    this._date = new realDate(timeNow);
                }

                return this._date;
            }

            private _date;

            public valueOf() {
                return this._date.valueOf();
            }
            public setHours(a,b,c,d) {
                return this._date.setHours(a,b,c,d);
            }
            public setDate(d) {
                return this._date.setDate(d);
            }
            public getDay() {
                return this._date.getDay();
            }
            public getDate() {
                return this._date.getDate();
            }
            public getMonth() {
                return this._date.getMonth();
            }
            public getTime() {
                return this._date.getTime();
            }
            public getFullYear() {
                return this._date.getFullYear();
            }
        };
    });

    afterAll(() => {
        global.Date = realDate;
    });

    it("provides constant timestamps", () => {
        const ts1 = Date.now();
        const ts2 = +new Date();
        expect(ts1).toEqual(ts2);
        expect(ts2).toEqual(timeNow.valueOf());
    });
    
    it("App should render ConfirmAlert correctly", () => {
        const wrapper = createMount()(
            <ConfirmAlert
                open={true}
                title="Title"
                message={<span>Test message</span>}
                handleYes={() => {}}
                handleNo={() => {}}
            />
        );

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("App should render AppAlert correctly", () => {
        const elem = document.createElement('div');
        elem.id = 'bottomAlert';

        document.body.appendChild(elem);
        
        const wrapper = createMount()(
            <AppAlert
                messages = {[{   id: 1, severity: 'error', text: 'Test3 message',}]}
            />
        );

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("App should render Performance correctly", () => {
        const storeFake = (state: any) => ({
            default: () => {},
            subscribe: () => {},
            dispatch: () => {},
            getState: () => ({ ...state })
        });

        const state = {
            performance: PerformanceStore.emptyState(),
        };

        const store = storeFake(state) as any;
        
        const elem = document.createElement('div');
        elem.id = 'footerToolbar';

        document.body.appendChild(elem);

        const wrapper = createMount()(<Provider store={store}>
            <MemoryRouter>
                    <Performance/>
            </MemoryRouter>
        </Provider>);

        expect(wrapper.html()).toMatchSnapshot();
    });
});
