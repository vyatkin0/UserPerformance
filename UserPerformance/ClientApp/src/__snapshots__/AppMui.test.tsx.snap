// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Material UI tests App should render AppAlert correctly 1`] = `"<div class=\\"MuiPaper-root MuiAlert-root MuiAlert-filledError AppAlert-alert-11 MuiPaper-elevation0\\" role=\\"alert\\"><div class=\\"MuiAlert-icon\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"></path></svg></div><div class=\\"MuiAlert-message\\">Test3 message</div></div>"`;

exports[`Material UI tests App should render ConfirmAlert correctly 1`] = `"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded\\" role=\\"dialog\\" aria-describedby=\\"alert-dialog-description\\" aria-labelledby=\\"alert-dialog-title\\"><div class=\\"MuiDialogTitle-root\\" id=\\"alert-dialog-title\\"><h2 class=\\"MuiTypography-root MuiTypography-h6\\">Title</h2></div><div class=\\"MuiDialogContent-root\\"><p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\" id=\\"alert-dialog-description\\"><span>Test message</span></p></div><div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary Mui-focusVisible Mui-focusVisible\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">No</span><span class=\\"MuiTouchRipple-root\\"><span class=\\"MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate\\" style=\\"width: 1px; height: 1px; top: -0.5px; left: -0.5px;\\"><span class=\\"MuiTouchRipple-child MuiTouchRipple-childPulsate\\"></span></span></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Yes</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;

exports[`Material UI tests App should render Performance correctly 1`] = `"<div class=\\"flex_container\\"><div><h1>Activities management</h1><h3></h3></div><div class=\\"period_control\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"#\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M14.71 6.71a.9959.9959 0 00-1.41 0L8.71 11.3c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L10.83 12l3.88-3.88c.39-.39.38-1.03 0-1.41z\\"></path></svg></a><span>1 - 31</span><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"#\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M9.29 6.71c-.39.39-.39 1.02 0 1.41L13.17 12l-3.88 3.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41L10.7 6.7c-.38-.38-1.02-.38-1.41.01z\\"></path></svg></a><hr class=\\"MuiDivider-root MuiDivider-flexItem MuiDivider-vertical\\"><div class=\\"period_control__select\\">December</div><hr class=\\"MuiDivider-root MuiDivider-flexItem MuiDivider-vertical\\"><div class=\\"period_control__select\\">2019</div><hr class=\\"MuiDivider-root MuiDivider-flexItem MuiDivider-vertical\\"><div class=\\"period_control__select\\">Today 01.01.2020</div></div></div><div class=\\"performance_container\\">Month performance<span class=\\"warning performance_data\\">0%</span>Working days<span class=\\"performance_data\\">0</span></div><div class=\\"MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded\\"><table class=\\"MuiTable-root Performance-table-15\\" aria-label=\\"simple table\\"><thead class=\\"MuiTableHead-root\\"><tr class=\\"MuiTableRow-root MuiTableRow-head\\"><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall\\" scope=\\"col\\">Name</th><th class=\\"MuiTableCell-root MuiTableCell-head Performance-monthHeadCell-19 MuiTableCell-alignCenter MuiTableCell-sizeSmall\\" scope=\\"col\\"></th><th class=\\"MuiTableCell-root MuiTableCell-head Performance-numCell-21 MuiTableCell-alignCenter MuiTableCell-sizeSmall\\" scope=\\"col\\">1.1</th></tr></thead><tbody class=\\"MuiTableBody-root\\"><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall\\" role=\\"cell\\" scope=\\"row\\">Day performance</th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall\\"></td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall\\"></td></tr><tr class=\\"MuiTableRow-root\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall\\" role=\\"cell\\" scope=\\"row\\">Spent hours</th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall\\"></td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall\\"><span class=\\"Performance-cellSpan-17\\">0</span></td></tr></tbody></table></div><div class=\\"Performance-divButtons-24\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\"><span class=\\"MuiButton-label\\">Save</span></button><a class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" aria-disabled=\\"false\\" href=\\"#/activities\\"><span class=\\"MuiButton-label\\">Select activities</span><span class=\\"MuiTouchRipple-root\\"></span></a></div>"`;
